pkg/backporter/backporter_test.go

```diff
@@ -135,79 +97,207 @@ func TestGetBugHandler(t *testing.T) {
+				expResponse = fmt.Sprintf(htmlPageStart, "Clones") + subPage + htmlPageEnd
+			} else {
+				expResponse = fmt.Sprintf(htmlPageStart, "Not Found") + subPage + htmlPageEnd
 			}
-			if resp := rr.Body.String(); resp != tc.results.htmlPage {
-				t.Errorf("Response differs from expected by: %s", diff.StringDiff(resp, tc.results.htmlPage))
+			if resp := rr.Body.String(); resp != expResponse {
+				t.Errorf("response differs from expected by: %s", diff.StringDiff(resp, expResponse))
 			}
 		})
 	}
 
 }
+
+func TestCreateCloneHandler(t *testing.T) {
+	fake := &bugzilla.Fake{}
+	fake.Bugs = map[int]bugzilla.Bug{}
+	fake.BugComments = map[int][]bugzilla.Comment{}
+
+	toBeClonedCreate := &bugzilla.BugCreate{
+		AssignedTo: "UnitTest",
+		Summary:    "Sample bug to test implementation of clones handler",
+	}
+	toBeClonedID, err := fake.CreateBug(toBeClonedCreate)
+	if err != nil {
+		t.Fatalf("error creating bug: %v", err)
+	}
+	clonedRelease := "4.4.z"
+	prunedReleaseSet := sets.NewString(allTargetVersions.List()...)
+	prunedReleaseSet.Delete(clonedRelease)
+
+	toBeCloned, err := fake.GetBug(toBeClonedID)
+	if err != nil {
+		t.Fatalf("error getting bug details from Fake.")
+	}
+	expectedCloneID := toBeClonedID + 1
+	testcases := []struct {
+		name       string
+		params     map[string]string
+		statusCode int
+		data       interface{}
+		tmplt      *template.Template
+	}{
+		{
+			"valid_parameters",
+			map[string]string{
+				"ID":      strconv.Itoa(toBeClonedID),
+				"release": clonedRelease,
+			},
+			http.StatusOK,
+			ClonesTemplateData{
+				toBeCloned,
+				[]*bugzilla.Bug{},
+				toBeCloned,
+				nil,
+				prunedReleaseSet.List(),
+				expectedCloneID,
+			},
+			clonesTemplate,
+		},
+		{
+			"bad_params",
+			map[string]string{
+				"ID":      "1000",
+				"release": "",
+			},
+			http.StatusNotFound,
+			"unable to fetch bug details- Bug#1000",
+			errorTemplate,
+		},
+	}
+	for _, tc := range testcases {
+		t.Run(tc.name, func(t *testing.T) {
+			formData := url.Values{}
+			formData.Set("ID", tc.params["ID"])
+			formData.Add("release", tc.params["release"])
+			req, err := http.NewRequest("POST", "/clones/create", bytes.NewBufferString(formData.Encode()))
+			req.Header.Set("Content-Type", "application/x-www-form-urlencoded; param=value")
+			if err != nil {
+				t.Fatal(err)
+			}
+			rr := httptest.NewRecorder()
+			handler := CreateCloneHandler(fake, allTargetVersions)
+			handler.ServeHTTP(rr, req)
+			if status := rr.Code; status != tc.statusCode {
+				t.Errorf("testcase '%v' failed: clonebug returned wrong status code - got %v, want %v", tc, status, tc.statusCode)
+			}
+
+			var buf bytes.Buffer
+			var pageStart string
+			if tc.statusCode == http.StatusOK {
+				pageStart = fmt.Sprintf(htmlPageStart, "Clones")
+				newClone, err := fake.GetBug(expectedCloneID)
+				if err != nil {
+					t.Fatalf("error while fetching clone details from mocked endpoint")
+				}
+				newClone.TargetRelease = []string{
```

Comment:
All properties of `newClone` are deterministic or not? If yes, can we just put it into the `templateData` and not mutate it during test execution by using a type assertion along the way?